src/decode.rs

```diff
@@ -5045,8 +5045,12 @@ pub unsafe fn rav1d_submit_frame(c: &mut Rav1dContext) -> Rav1dResult {
         }
     }
 
-    if (*f.dsp).ipred.intra_pred[DC_PRED as usize].is_none() {
+    // TODO(kkysen) Rather than lazy initializing this,
+    // we should probably initialize all the fn ptrs
+    // when `c` is allocated during [`rav1d_open`].
+    if !(*f.dsp).ipred.initialized {
```

Comment:
Is there a reason this should be in the ipred structure rather than the top-level DSP context? We are initializing all of it afaict?

Reply:
Hmm, yeah, that's probably better.  I was just confused since the C was checking `ipred`, but that was probably just done randomly.  I'll move it.

Reply:
Changed in [7d0c2ce..4609590](https://github.com/memorysafety/rav1d/compare/7d0c2ce6da7544f7a9f026d1b44478f4ac19b5ff..4609590678b588824c7770f5d05ba378e2914709).